homeassistant/components/mqtt/client.py

```diff
@@ -309,12 +308,13 @@ def setup(self) -> None:
         if (client_id := config.get(CONF_CLIENT_ID)) is None:
             # PAHO MQTT relies on the MQTT server to generate random client IDs.
             # However, that feature is not mandatory so we generate our own.
-            client_id = mqtt.base62(uuid.uuid4().int, padding=22)
+            client_id = None  # mqtt.base62(uuid.uuid4().int, padding=22)
```

Comment:
```suggestion
            client_id = None
```
We could allow the client library to generate an id. The `base62` method was no longer exported, so this needed to changed.